src/_pytest/python.py

```diff
@@ -231,16 +229,12 @@ def pytest_pycollect_makeitem(
                 lineno=lineno + 1,
             )
         elif getattr(obj, "__test__", True):
-            if is_generator(obj):
-                res = Function.from_parent(collector, name=name)
-                reason = (
-                    f"yield tests were removed in pytest 4.0 - {name} will be ignored"
+            if inspect.isgeneratorfunction(obj):
+                raise RuntimeError(
+                    "'yield' keyword is allowed in fixtures, but not in tests ({name})"
```

Comment:
An internal error will unfortunately show an internal traceback which is quite noisy. Let's use `fail` instead (also it was missing the `f` for the f-string):

```suggestion
                fail(
                    f"'yield' keyword is allowed in fixtures, but not in tests ({name})",
                    pytrace=False,
```